src/components/gas/GasSpeedButton.js

```diff
@@ -0,0 +1,569 @@
+      color={
+        theme === 'dark'
+          ? colors.alpha(darkModeThemeColors.blueGreyDark, 0.6)
+          : colors.alpha(colors.blueGreyDark, 0.6)
+      }
+      size="lmedium"
+      weight="bold"
+    >
+      {formatter()}
+    </Label>
+  );
+};
+
+const GasSpeedButton = ({
+  showGasOptions = false,
+  testID,
+  theme = 'dark',
+  options = null,
+  currentNetwork,
+  asset,
+  horizontalPadding = 20,
+}) => {
+  const { colors } = useTheme();
+  const { navigate, goBack } = useNavigation();
+  const { nativeCurrencySymbol, nativeCurrency } = useAccountSettings();
+  const colorForAsset = useColorForAsset(asset || {});
+
+  // if ETH color, use blueApple
+  const assetColor = useMemo(() => {
+    if (colorForAsset === colors.brighten(lightModeThemeColors.dark)) {
+      return colors.appleBlue;
+    }
+    return colorForAsset;
+  }, [colorForAsset, colors]);
+
+  const {
+    gasFeeParamsBySpeed,
+    gasFeesBySpeed,
+    isSufficientGas,
+    updateGasFeeOption,
+    selectedGasFee,
+    selectedGasFeeOption,
+  } = useGas();
+
+  const [estimatedTimeValue, setEstimatedTimeValue] = useState(0);
+  const [estimatedTimeUnit, setEstimatedTimeUnit] = useState('min');
+  const [shouldOpenCustomGasSheet, setShouldOpenCustomGasSheet] = useState(
+    false
+  );
+  const prevShouldOpenCustomGasSheet = usePrevious(shouldOpenCustomGasSheet);
+
+  // Because of the animated number component
+  // we need to trim the native currency symbol
+  // (and leave the number only!)
+  // which gets added later in the formatGasPrice function
+  const price = useMemo(() => {
+    const gasPrice = get(
+      selectedGasFee,
+      `gasFee.estimatedFee.native.value.display`
+    );
+    const price = (isNil(gasPrice) ? '0.00' : gasPrice)
+      .replace(',', '') // In case gas price is > 1k!
+      .replace(nativeCurrencySymbol, '')
+      .trim();
+    return price;
+  }, [nativeCurrencySymbol, selectedGasFee]);
+
+  const isL2 = useMemo(() => isL2Network(currentNetwork), [currentNetwork]);
+
+  const formatGasPrice = useCallback(
+    animatedValue => {
+      // L2's are very cheap,
+      // so let's default to the last 2 significant decimals
+      if (isL2) {
+        const numAnimatedValue = Number.parseFloat(animatedValue);
+        if (numAnimatedValue < 0.01) {
+          return `${nativeCurrencySymbol}${numAnimatedValue.toPrecision(2)}`;
+        } else {
+          return `${nativeCurrencySymbol}${numAnimatedValue.toFixed(2)}`;
+        }
+      } else {
+        return `${nativeCurrencySymbol}${
+          nativeCurrency === 'ETH'
+            ? (Math.ceil(Number(animatedValue) * 10000) / 10000).toFixed(4)
+            : (Math.ceil(Number(animatedValue) * 100) / 100).toFixed(2)
+        }`;
+      }
+    },
+    [isL2, nativeCurrencySymbol, nativeCurrency]
+  );
+
+  const gasIsNotReady = useMemo(
+    () =>
+      isEmpty(gasFeeParamsBySpeed) ||
+      isEmpty(selectedGasFee?.gasFee) ||
+      typeof isSufficientGas === 'undefined',
```

Comment:
`isSufficientGas` default value has been changed to `null` when it was previously `undefined` so this check will now fail

Reply:
same on line 794 of `TransactionConfirmationScreen` when it does the check: `if (!isMessage && !isBalanceEnough && isSufficientGas === undefined)`

Reply:
same on line 799 of `TransactionConfirmationScreen`

Reply:
ideally we don't use boolean and null/undefined state in this way as it can be very confusing to maintain and instead use a different metric for when the gas isn't ready - I had done this in a different branch, can see if I can dig it up

Reply:
hey this still needs to be addressed to do `=== null` in `TransactionConfirmationScreen` where it is currently just checking against `undefined`